### PR TITLE
Reduce requirements for calibrate microservice

### PR DESCRIPTION
**Pull request by bobturneruk**
# Hotfix

**_WHAT DOES THIS PR DO?_**

Enables the calibrate microservice to be built, run and tested using docker, with a (hopefully) reduced requirements load.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**

https://airqoteam.atlassian.net/browse/PLAT-523

https://airqoteam.atlassian.net/browse/PLAT-526 

**_WHAT IS THE LINK TO THE PR BRANCH_**

https://github.com/airqo-platform/AirQo-api/tree/calibrate-req

**_HOW DO I TEST OUT THIS PR?_**

Follow instructions in `src/calibrate/readme.md`

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

See `src/calibrate/readme.md`

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

No

**_ARE THERE ANY RELATED PRs?_**

https://github.com/airqo-platform/AirQo-api/pull/326

### CODE REVIEW

File: src/calibrate/readme.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+```{json}
+{
+    "datetime": "2020-07-15 13:00:00",
+    "raw_values": [
+        {
+            "device_id": "aq_01",
+            "pm2.5": 44.12,
+            "pm10": 54.20,
+            "temperature": 25.3,
```
Comment (Baalmart) [inline src/calibrate/readme.md:32]: This is great work! Just sharing some data engineering notes/thoughts about how this endpoint will be used within our data pipeline.

I can imagine that we have to first use the device ID or channel ID to search for the temperature and humidity readings for each device and then create a body to be used for this endpoint. If we use the channel ID, then we would also have to use the GET Devices endpoint to search for device name of each channel ID.

A few more questions/thoughts...

1. One thing to note is that there might be a good number of devices without temperature and humidity readings, what do we do for them?
2. Shall we be using the recent transform endpoint to read the temperature and humidity readings?
3. In this README, could we also showcase a sample of the response message after making the POST?

Thanks.

**Review by Baalmart — COMMENTED** [review]
Just curious...from the PR description, where in the changed files are we "`enabling this microservice to run with Docker with a reduced requirements load`"? @bobturneruk

File: src/calibrate/requirements/prod.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Flask==1.1.2
```
Comment (ja-odur) [inline src/calibrate/requirements/prod.txt:1]: These two requirement files (dev.txt, prod.txt) are duplicate. Probably we could stick with one for easy management.

**Comment by bobturneruk** [comment]
> Just curious...from the PR description, where in the changed files are we "enabling this microservice to run with Docker with a reduced requirements load"? @bobturneruk

- Current requirements are reduced from previous src/calibrate/requirements.txt.
- Docker build previously failed due to referencing a non-existent requirements file.

File: src/calibrate/requirements/prod.txt (hunk 2 above)
Comment (bobturneruk) [reply src/calibrate/requirements/prod.txt:1]: Makes sense - I was following the format of other modules. Perhaps we could change this later. @ja-odur - please let me know if you require a change for approval.

File: src/calibrate/readme.md (hunk 1 above)
Comment (bobturneruk) [reply src/calibrate/readme.md:32]: @Baalmart - I'm documenting how it is at the moment. These are good points. Perhaps they could become a Jira issue for future resolution as some effort would be involved to implement.

File: src/calibrate/requirements/prod.txt (hunk 2 above)
Comment (ja-odur) [reply src/calibrate/requirements/prod.txt:1]: Yes, we could fix this later

Comment (bobturneruk) [reply src/calibrate/requirements/prod.txt:1]: Great. Are you happy to approve @ja-odur?

File: src/calibrate/readme.md (hunk 1 above)
Comment (bobturneruk) [reply src/calibrate/readme.md:32]: @Priscilla-AL - can you shed any light on this? I did not intend to change the behaviour of the microservice (only to reduce image size by having fewer requirements), but we have no version on staging to compare with so we're not sure if the output is correct in all circumstances.

Comment (bobturneruk) [reply src/calibrate/readme.md:32]: I have raised https://airqoteam.atlassian.net/browse/PLAT-526 to address this later, seperately from the change to requirements. If this approach is OK, I suggest the PR can be approved.

File: src/calibrate/requirements/prod.txt (hunk 2 above)
Comment (bobturneruk) [reply src/calibrate/requirements/prod.txt:1]: I will change to a single `requirements.txt` file.

**Review by ja-odur — APPROVED** [review]
LGTM

**Review by lilly-muyama — APPROVED** [review]
LGTM

**Comment by bobturneruk** [comment]
Thanks @lilly-muyama and @ja-odur - just need one more!

**Review by sserurich — APPROVED** [review]
This works well, only caveat is explicitly specifying  this in code "airqo-250220-642e046f4223.json in regression.py. It's not part of the changes, but we're better off reading that from env variables.

**Comment by bobturneruk** [comment]
Thanks @sserurich - I've raised https://airqoteam.atlassian.net/browse/PLAT-527 in response to your comment.

**Pull request merged**